Common/TableProducer/PID/pidTPCFull.cxx

```diff
@@ -199,11 +178,14 @@ struct tpcPidFull {
             LOG(fatal) << "Local path must be set (flag networkPathLocally)! Aborting...";
           }
           LOG(info) << "Using local file [" << networkPathLocally.value << "] for the TPC PID response correction.";
-          Network temp_net(networkPathLocally.value,
-                           enableNetworkOptimizations.value,
-                           activeThreads);
+          OnnxModel temp_net(networkPathLocally.value);
           network = temp_net;
-          network.evalNetwork(std::vector<float>(network.getInputDimensions(), 1.)); // This is an initialisation and might reduce the overhead of the model
+          if (enableNetworkOptimizations) {
+            Ort::SessionOptions* sessOpts = network.getSessionOptions();
```

Comment:
another point for the future - maybe we can wrap this optimization part as well into one-liner for the users.

Reply:
That could be done easily, yes. I also had it in my header as a boolean in the constructor. I just thought that for full flexibility with all the options ONNX provides (thread / CPU / GPU settings or model optimizations), it is easiest if one can modify the sessionOptions and then reset the sessions with the optimizations without being limited by a function. But we can surely also put it in a function to make it a bit easier for the users.

Reply:
ok, yeah, that is open for discussion. There can be a wrapper with an optional argument to select optimization type, for example. That could balance flexibility and simplicity.